### PR TITLE
Implement screen search command

### PR DESCRIPTION
**Pull request by sadeghhadjigi**

### CODE REVIEW

File: i8_terminal/commands/screen/screen_search.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    export_path: Optional[str],
+    sort_by: Optional[str],
+    sort_order: Optional[str],
+) -> None:
+    console = Console()
+    if not metrics and not view_name:
+        console.print("The 'metrics' or 'view_name' parameter must be provided", style="yellow")
+        return
+    if view_name and metrics:
+        console.print("The 'metrics' or 'view_name' options are mutually exclusive", style="yellow")
```
Comment (panteaHK) [inline i8_terminal/commands/screen/screen_search.py:79]: Provide a message with clear instructions for the user at the end to tell them how to fix the error. For example: "Provide a value only for one of the two."

File: i8_terminal/commands/screen/screen_search.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+) -> Tuple[List[str], Optional[pd.DataFrame]]:
+    max_count = 20
+    for index, condition in enumerate(conditions):
+        condition_parts = condition.split(":")
+        metric = condition_parts[0]
+        metric_parts = metric.split(".")
+        if len(metric_parts) == 1:
+            metric_default_period_type = get_metric_info(metric_parts[0])["default_period_type"]
+            period_type = (
+                ".q" if metric_default_period_type == "Q" else ".fy" if metric_default_period_type == "FY" else ""
```
Comment (investoreight) [inline i8_terminal/commands/screen/screen_search.py:35]: The default period type should be "d" (not empty string)

File: i8_terminal/common/metrics.py [hunk 3]
```diff
@@ -74,6 +74,24 @@ def get_all_metrics_df() -> DataFrame:
     return df
 
 
+def get_all_metrics_screening_conditions_df() -> DataFrame:
```
Comment (investoreight) [inline i8_terminal/common/metrics.py:77]: This is no longer required. Please delete it.

File: i8_terminal/types/condition_param_type.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+from i8_terminal.common.formatting import format_number_v2
+from i8_terminal.common.metrics import get_all_metrics_df
+from i8_terminal.types.auto_complete_choice import AutoCompleteChoice
+
+PERIOD_TYPES: Dict[str, str] = {
+    "fy": "mry",
+    "q": "mrq",
+    "ttm": "ttm",
+    "ytd": "ytd",
+    "d": "",
```
Comment (investoreight) [inline i8_terminal/types/condition_param_type.py:13]: With the new changes to the ES index, we now always have period type. d => d, r => r.

**Review by investoreight — APPROVED** [review]

**Comment by investoreight** [comment]
@sadeghhadjigi Please address the above three comments in a separate PR.

**Pull request merged**